crates/weaver_live_check/src/lib.rs

```diff
@@ -337,5 +341,9 @@ impl LiveCheckStatistics {
 /// Samples implement this trait to run live checks on themselves
 pub trait LiveCheckRunner {
     /// Run the live check
-    fn run_live_check(&mut self, live_checker: &mut LiveChecker, stats: &mut LiveCheckStatistics);
+    fn run_live_check(
+        &mut self,
+        live_checker: &mut LiveChecker,
+        stats: &mut LiveCheckStatistics,
+    ) -> Result<(), Error>;
```

Comment:
it's still a rust complain @jsuereth, if the result with error was more like an exception, it would take an effort to ignore it 😅

Reply:
It did take effort to ignore it.  All the implementations of this trait were explicitly taking the ok state and ignoring the error state of the result.

Our clippy would fire if someone hadn't explicitly decided to do that and tried to implicitly ignore errors.

This is similar to the issue of "catching too much" with checked exceptions where decide to catch everything rather than think through what to propagate.

The only complaint is have on rust is that "if let Ok(...) =" should probably be something we scrutinize more in code review as the equivalent of ignoring exceptions via "try { } catch(anything) { ...ignore...}"